### PR TITLE
Handling events

### PR DESCRIPTION
**Pull request by thneves**
Add click Events to make calculations on the display
Implement final styling

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -1,19 +1,38 @@
...
+      Calculate(this.state, button),
+    );
+  }
+
+  render() {
+    return (
       <div className="calculator">
-        <Display result={Calculate(null, null)} />
-        <Panel />
+        {/* eslint-disable-next-line react/destructuring-assignment, max-len */}
```
Comment (addod19) [inline src/components/App.js:30]: Kindly remove this comment!!! Let us avoid the ESlint inline disabling rule

File: src/components/Panel.js [hunk 2]
```diff
@@ -1,37 +1,48 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import Button from './Button';
 import '../styles/Panel.css';
 
-function Panel() {
+function Panel({ clickHandler }) {
```
Comment (addod19) [inline src/components/Panel.js:6]: Kindly use the ES6 arrow syntax instead of the function keyword!!!

File: src/components/Panel.js [hunk 3]
```diff
@@ -1,37 +1,48 @@
...
+        </div>
+        <div className="panel-row">
+          <Button name="0" clickHandler={() => clickHandler('0')} />
+          <Button name="." clickHandler={() => clickHandler('.')} />
+          <Button name="=" clickHandler={() => clickHandler('=')} />
         </div>
       </div>
     </>
   );
 }
```
Comment (addod19) [inline src/components/Panel.js:43]: Kindly implement the handleClick function. For example; 
```
handleClick = (buttonName) => {
  this.props.clickHandler(buttonName);
};
```
It should look like this :smile:

File: src/components/Button.js [hunk 4]
```diff
@@ -2,17 +2,18 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import '../styles/Button.css';
 
-function Button(props) {
-  const { name } = props;
-  return <button className="btn" type="button">{name}</button>;
+function Button({ name, clickHandler }) {
```
Comment (addod19) [inline src/components/Button.js:5]: Kindly use the ES6 arrow syntax instead of the function keyword :smile:

File: src/components/Display.js [hunk 5]
```diff
@@ -2,22 +2,24 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import '../styles/Display.css';
 
-function Display(props) {
-  const { result } = props;
+function Display({ result, next, operation }) {
```
Comment (addod19) [inline src/components/Display.js:5]: Let us kindly use the ES6 arrow syntax instead of the function keyword :+1:

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @thneves, congratulations on this milestone :fireworks: The App component has been converted into a class component :+1: Its state is initialized in the constructor :1st_place_medal: HandleClick also implemented in the constructor :slightly_smiling_face: 
However, there are a few comments we need to address before proceeding :+1: 

Kindly check the comments below:

Once these are addressed, kindly submit another code review request :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Review by Forison — CHANGES_REQUESTED** [review]
## Status::Approved❌

# 👋  @thneves
 
#### Good job on addressing the issue raised in the previous review, functions are defined with arrow function, eslint rule have been set again. However the readme and calculator have some flaws which need addressing🤗

* [ ] The calculator throws into error when we devide a number by zero
- let us return undefined for such computation ie `5/0=undefined`

* [ ] Let us add an image of the calculator to the readme

* [ ] Let us make sure the live link reflect the actual app, as seen in our local environment

* [ ] Let remove information that are not reflected in the project in the readme, we have a command for `test` but there is no test, at least not at this moment.

#### That is it, kindly make the changes and submit a re-review👨‍🔧

#### Happy coding👨‍💻

Addo Forison

**Review by Forison — APPROVED** [review]
## Status::Approved✔️

# Hello @thneves 

#### Thank you for addressing the issues raised earlier, kindly go ahead and merge the branch and proceed to your next task🤗🤗🤗

#### Happy coding👨‍💻

Addo Forison

**Pull request merged**